packages/core/src/entity/EntityFactory.ts

```diff
@@ -103,12 +103,15 @@ export class EntityFactory {
   /**
    * denormalize PK to value required by driver (e.g. ObjectId)
    */
-  private denormalizePrimaryKey<T extends AnyEntity<T>>(data: EntityData<T>, primaryKey: string): void {
+  private denormalizePrimaryKey<T extends AnyEntity<T>>(data: EntityData<T>, primaryKey: string, metadata: EntityProperty ): void {
```

Comment:
```suggestion
  private denormalizePrimaryKey<T extends AnyEntity<T>>(data: EntityData<T>, primaryKey: string, prop: EntityProperty<T>): void {
```

please use `prop` as the variable name, as it holds metadata for single property (for metadata of single entity we use `meta`, and `metadata` is reserved for the instance of `MetadataStorage`, so all entities metadata).